src/io/perun.clj

```diff
@@ -36,50 +36,48 @@
 
 (deftask markdown
   "Parse markdown files"
-  [d datafile        DATAFILE        str "Target datafile with all parsed meta information"
-   f create-filename CREATE_FILENAME str "Function that creates final target filename of the file"]
+  [f create-filename CREATE_FILENAME str "Function that creates final target filename of the file"]
   (let [pod (create-pod markdown-deps)
         tmp (boot/tmp-dir!)
         options (merge +markdown-defaults+ *opts*)]
     (boot/with-pre-wrap fileset
-      (let [markdown-files (->> fileset boot/user-files (boot/by-ext ["md" "markdown"]) (map #(.getPath (boot/tmp-file %))))]
-        (pod/with-call-in @pod
-          (io.perun.markdown/parse-markdown
-            ~(.getPath tmp)
-            ~options
-            ~markdown-files))
-        (commit fileset tmp)))))
+      (let [markdown-files (->> fileset boot/user-files (boot/by-ext ["md" "markdown"]) (map #(.getPath (boot/tmp-file %))))
+            parsed-metadata (pod/with-call-in @pod
+                              (io.perun.markdown/parse-markdown
+                                ~(.getPath tmp)
+                                ~options
+                                ~markdown-files))
+            fs-with-meta (with-meta fileset {:metadata parsed-metadata})]
+        (u/info "Parsed markdown files\n")
+        (commit fs-with-meta tmp)))))
```

Comment:
Nope. Commit is to write changes to disk. Clojure metadata is just associated with Java object.